npm/ux/circuit.tsx

```diff
@@ -67,31 +68,39 @@ export function Circuit(props: { circuit: qviz.Circuit }) {
 }
 
 /* This component is exclusive to the VS Code panel */
-export function CircuitPanel(props: {
-  title: string;
-  subtitle: string;
-  circuit?: qviz.Circuit;
-  errorHtml?: string;
-}) {
+export function CircuitPanel(props: CircuitProps) {
+  const error = props.errorHtml ? (
+    <div>
+      <p>
+        {props.circuit
+          ? "The program encountered a failure. See the error(s) below."
+          : "A circuit could not be generated for this program. See the error(s) below."}
+        <br />
+      </p>
+      <div dangerouslySetInnerHTML={{ __html: props.errorHtml }}></div>
+    </div>
+  ) : null;
+
   return (
     <div class="qs-circuit-panel">
       <div>
         <h1>{props.title}</h1>
       </div>
       {props.circuit ? <Circuit circuit={props.circuit}></Circuit> : null}
-      <div class="qs-circuit-error">
-        {props.errorHtml ? (
-          <div>
-            <p>
-              A circuit could not be generated for this program. See the
-              error(s) below.
-              <br />
-            </p>
-            <div dangerouslySetInnerHTML={{ __html: props.errorHtml }}></div>
-          </div>
-        ) : null}
-      </div>
-      <p>{props.subtitle /* target profile */}</p>
+      <div class="qs-circuit-error">{error}</div>
+      <p>{props.targetProfile /* target profile */}</p>
```

Comment:
I'm not sure the comment here is adding a lot of clarity :-)

Reply:
Let me clarify that for you: This parameter is the Target Profile. :P